Tests/AmplitudeTests.m

```diff
@@ -637,123 +617,122 @@ -(void)testSetOfflineTruncate {
-    XCTAssertEqual([dbHelper getTotalEventCount], 6);
-
+    
+    // when setOffline:YES, all events should only in buffer
+    XCTAssertEqual([self.amplitude getEventCount], 0);
+    XCTAssertEqual([self.amplitude getIdentifyCount], 0);
+    XCTAssertEqual([self.amplitude.eventsBuffer count], 3);
+    XCTAssertEqual([self.amplitude.identifyBuffer count], 3);
+    
+    [self.amplitude setOffline:NO];
     [self.amplitude logEvent:@"test4"];
     [self.amplitude identify:[[AMPIdentify identify] unset:@"key4"]];
     [self.amplitude flushQueue];
 
-    XCTAssertEqual([dbHelper getEventCount], 3);
-    XCTAssertEqual([dbHelper getIdentifyCount], 3);
-    XCTAssertEqual([dbHelper getTotalEventCount], 6);
-
-    NSMutableArray *events = [dbHelper getEvents:-1 limit:-1];
+    XCTAssertEqual([self.amplitude.eventsBuffer count], 3);
+    XCTAssertEqual([self.amplitude.identifyBuffer count], 3);
+    
+    NSMutableArray *events = self.amplitude.eventsBuffer;
     XCTAssertEqual([events count], 3);
     XCTAssertEqualObjects([events[0] objectForKey:@"event_type"], @"test2");
     XCTAssertEqualObjects([events[1] objectForKey:@"event_type"], @"test3");
     XCTAssertEqualObjects([events[2] objectForKey:@"event_type"], @"test4");
 
-    NSMutableArray *identifys = [dbHelper getIdentifys:-1 limit:-1];
+    NSMutableArray *identifys = self.amplitude.identifyBuffer;
     XCTAssertEqual([identifys count], 3);
     XCTAssertEqualObjects([[[identifys[0] objectForKey:@"user_properties"] objectForKey:@"$unset"] objectForKey:@"key2"], @"-");
     XCTAssertEqualObjects([[[identifys[1] objectForKey:@"user_properties"] objectForKey:@"$unset"] objectForKey:@"key3"], @"-");
     XCTAssertEqualObjects([[[identifys[2] objectForKey:@"user_properties"] objectForKey:@"$unset"] objectForKey:@"key4"], @"-");
-
-
-    [self.amplitude setOffline:NO];
-    [self.amplitude flushQueue];
-
-    XCTAssertEqual([dbHelper getEventCount], 0);
-    XCTAssertEqual([dbHelper getIdentifyCount], 0);
-    XCTAssertEqual([dbHelper getTotalEventCount], 0);
 }
 
 -(void)testTruncateEventsQueues {
     int eventMaxCount = 50;
+
     XCTAssertGreaterThanOrEqual(eventMaxCount, kAMPEventRemoveBatchSize);
     self.amplitude.eventMaxCount = eventMaxCount;
 
-    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
-    [self.amplitude setOffline:YES];
     for (int i = 0; i < eventMaxCount; i++) {
         [self.amplitude logEvent:@"test"];
     }
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount);
+    XCTAssertEqual([self.amplitude.eventsBuffer count], eventMaxCount);
 
     [self.amplitude logEvent:@"test"];
     [self.amplitude flushQueue];
-    XCTAssertEqual([dbHelper getEventCount], eventMaxCount - (eventMaxCount/10) + 1);
+    XCTAssertEqual([self.amplitude.eventsBuffer count], eventMaxCount - (eventMaxCount/10) + 1);
 }
 
 -(void)testTruncateEventsQueuesWithOneEvent {
+    [self.amplitude setEventUploadThreshold:1];
```

Comment:
I didn't set the default value, because the cases are different in test cases, it depends on how many events you logged before you flush the queue. So I have the setEventUploadTreshold set in each test case.